cmd/registry/controller/local-tasks.go

```diff
@@ -47,16 +65,38 @@ func (task *ExecCommandTask) Run(ctx context.Context) error {
 		"taskID": fmt.Sprintf("{%s}", task.TaskID),
 	})
 
-	if strings.HasPrefix(task.Action.Command, "resolve") {
-		logger.Debug("Failed Execution: 'resolve' not allowed in action")
-		return errors.New("'resolve' not allowed in action")
+	if strings.HasPrefix(task.Action.Command, "registry resolve") {
+		logger.Debug("Failed Execution: 'registry resolve' not allowed in action")
+		return errors.New("'registry resolve' not allowed in action")
 	}
 
-	cmd := exec.Command("registry", strings.Fields(task.Action.Command)...)
-	cmd.Stdout, cmd.Stderr = os.Stdout, os.Stderr
-	if err := cmd.Run(); err != nil {
-		logger.WithError(err).Debug("Failed Execution: failed running command")
-		return errors.New("failed running command")
+	// first party registry commands
+	if strings.HasPrefix(task.Action.Command, "registry") {
+		fullCmd := strings.Fields(fmt.Sprintf("%s --task_id=%s", task.Action.Command, task.TaskID))
```

Comment:
Optional: `append(strings.Fields(task.Action.Command), fmt.Sprintf("--task_id=%s", task.TaskID))` feels a bit clearer to me.

Reply:
Done